reducers/plugins/index.js

```diff
@@ -136,10 +168,18 @@ function components(state = {}, action) {
     case ActionTypes.RECEIVED_PLUGIN_COMPONENT: {
         if (action.name && action.data) {
             const nextState = {...state};
-            const currentArray = nextState[action.name] || [];
-            const nextArray = [...currentArray];
+            const currentArray = (nextState[action.name] || []).filter((c) => !c.subMenu);
+            let actionData = action.data;
+            if (action.name === 'PostDropdownMenu' && actionData.parentMenuId) {
+                const subMenu = (nextState[action.name] || []).find((c) => c.subMenu && c.pluginId === actionData.pluginId);
+                actionData = buildSubMenu(subMenu, action.data);
+                if (!actionData) {
```

Comment:
i don't think we're ever returning a `falsy` value from `buildSubMenu` so this case will never be hit. What we want is that if an action is dispatched for a menu that doesn't exist, or was a bad action, we don't want to update anything but return the original state.

Reply:
should be handled by the above logic